packages/uploadthing/src/internal/handler.ts

```diff
@@ -62,43 +62,63 @@ const createUTFetch = (
     return response;
   };
 };
-
-const fileCountLimitHit = (
+const fileCountBoundsCheck = (
   files: { name: string }[],
   routeConfig: ExpandedRouteConfig,
 ) => {
   const counts: Record<string, number> = {};
+  const results: Record<string, any> = {
+    minCountHit: false,
+    limitHit: false,
+  };
 
   files.forEach((file) => {
     const type = getTypeFromFileName(file.name, objectKeys(routeConfig));
-
-    if (!counts[type]) {
-      counts[type] = 1;
-    } else {
-      counts[type] += 1;
-    }
+    counts[type] = (counts[type] || 0) + 1;
   });
 
   for (const _key in counts) {
     const key = _key as FileRouterInputKey;
     const count = counts[key];
-    const limit = routeConfig[key]?.maxFileCount;
+    const min = routeConfig[key]?.minFileCount;
+    const max = routeConfig[key]?.maxFileCount;
 
-    if (!limit) {
+    if (typeof min !== "number" || typeof max !== "number") {
       logger.error(routeConfig, key);
+      const configKey =
+        typeof min !== "number" ? "minFileCount" : "maxFileCount";
+      throw new UploadThingError({
+        code: "BAD_REQUEST",
+        message: `Invalid config during file count - missing ${configKey}`,
+        cause: `Expected route config to have a ${configKey} for key ${key} but none was found.`,
+      });
+    }
+
+    if (min > max) {
       throw new UploadThingError({
         code: "BAD_REQUEST",
-        message: "Invalid config during file count",
-        cause: `Expected route config to have a maxFileCount for key ${key} but none was found.`,
+        message:
+          "Invalid config during file count - minFileCount > maxFileCount",
+        cause: `minFileCount should be less than maxFileCount for key ${key}. got: ${min} > ${max}`,
       });
     }
 
-    if (count > limit) {
-      return { limitHit: true, type: key, limit, count };
+    if (count < min) {
+      results.minCountHit = true;
+      results.type = key;
+      results.minCount = min;
+      results.count = count;
+    }
+
+    if (count > max) {
+      results.limitHit = true;
+      results.type = key;
+      results.limit = max;
+      results.count = count;
     }
   }
 
-  return { limitHit: false };
+  return results;
```

Comment:
micro-optimization: we can just return early in the for loop to skip extra iterations